graph.py

```diff
@@ -0,0 +1,71 @@
+from itertools import product
+import random
+from collections import deque
+
+class Vertex():
+    def __init__(self,name:str,value=None,children:list=None):
+        if name is not None:
```

Comment:
It makes the definition of this data-type a bit loose to have `name` defined sometimes but not all the time.
```python
self.name = name
if not name:
    raise GraphError("Name is a required field")
```